projects/packages/wp-js-data-sync/src/class-data-sync.php

```diff
@@ -134,27 +136,21 @@ public function add_to_plugin_page( $plugin_page, $parent_page ) {
 	 *
 	 * @return Data_Sync - A new instance of the Data_Sync class.
 	 */
-	public static function setup( $registry_name, $script_handle, $plugin_page = null, $parent_page = 'admin' ) {
-
-		$registry = Registry::get_instance( $registry_name );
-
-		/**
-		 * The plugin page slug can be anything, but makes setup easier to read by making assumptions.
-		 * This assumes that the plugin page string is going to match the registry namespace,
-		 * formatted as a http parameter. (kebab case)
-		 *
-		 * Example:
-		 * Registry with namespace:  `jetpack_boost` should be
-		 * automatically attached to `admin.php?page=jetpack-boost`
-		 */
-		if ( $plugin_page === null ) {
-			$plugin_page = $registry->get_namespace_http();
-		}
+	public function attach_to_plugin( $script_handle, $plugin_page_hook ) {
+		$this->script_handle = $script_handle;
+		add_action( $plugin_page_hook, array( $this, '_print_options_script_tag' ) );
+	}
 
-		$instance = new self( $script_handle, $registry );
-		$instance->add_to_plugin_page( $plugin_page, $parent_page );
+	public function get_registry() {
+		return $this->registry;
+	}
 
-		return $instance;
+	public function register( $key, $schema, $entry = null ) {
+		if ( ! $entry instanceof Data_Sync_Entry_Adapter ) {
```

Comment:
`Data_Sync_Entry` is an instance of `Data_Sync_Entry_Adapter` and entry is just supposed to be an instance of `Entry_Can_Get` based on what I understand. Is this check wrong?